src/main/java/cms/gov/madie/measure/resources/MeasureController.java

```diff
@@ -191,14 +192,23 @@ public ResponseEntity<List<AclSpecification>> updateAccessControl(
       @RequestBody @Validated AclOperation aclOperation,
       @Value("${admin-api-key}") String apiKey) {
     List<AclSpecification> aclSpecifications =
-        measureService.updateAccessControlList(id, aclOperation);
+        measureService.updateAccessControlList(id, aclOperation, "admin");
     return ResponseEntity.ok().body(aclSpecifications);
   }
 
   @GetMapping("/measures/shared")
-  public ResponseEntity<Map<String, List<String>>> getSharedWithUserIds(
+  public ResponseEntity<Map<String, List<SharedUser>>> getSharedMeasures(
       HttpServletRequest request, @RequestParam(name = "measureIds") List<String> measureIds) {
-    return ResponseEntity.ok().body(measureService.getSharedWithUserIds(measureIds));
+    return ResponseEntity.ok().body(measureService.getSharedMeasures(measureIds));
+  }
+
+  @PutMapping("/measures/shared")
+  public ResponseEntity<Map<String, List<AclSpecification>>> updateSharedMeasures(
+      HttpServletRequest request,
+      @RequestBody Map<String, List<String>> measures,
```

Comment:
don't think this is a map of measures. maybe measureUserIdMap or something similar?